apis/placement/v1alpha1/stagedupdate_types.go

```diff
@@ -0,0 +1,441 @@
+
+	// The status of the post update tasks that are associated with current stage.
+	// +kubebuilder:validation:Optional
+	AfterStageTaskStatus []AfterStageTaskStatus `json:"currentPostStageTaskStatus,omitempty"`
+}
+
+// ClusterUpdateStatus defines the status of the update run on a cluster.
+type ClusterUpdateStatus struct {
+	// The name of the cluster.
+	// +kubebuilder:validation:Required
+	ClusterName string `json:"clusterName"`
+
+	// The status of the update run in the stage.
+	// +kubebuilder:validation:Required
+	Status UpdateStatus `json:"status"`
+}
+
+// UpdateStatus defines the status of a resource in an update run.
+type UpdateStatus struct {
+	// The status of the update on the cluster.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=NotStarted;Running;Completed;Failed;WaitingForApproval
+	State UpdateState `json:"state"`
+
+	// message is a human readable message indicating details about the update status.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MaxLength=1024
+	Message string `json:"message,omitempty"`
+
+	// The time when the update started on the resource.
+	// +kubebuilder:validation:Optional
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+
+	// The time when the update completed on the resource.
+	// +kubebuilder:validation:Optional
+	CompletedTime *metav1.Time `json:"completedTime,omitempty"`
+
+	// The error detail if the update failed on the resource.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MaxLength=4096
+	ErrorDetail string `json:"errorDetail,omitempty"`
+}
+
+// UpdateState identifies the state of resource in an update run.
+// +enum
+type UpdateState string
+
+const (
+	// UpdateStateNotStarted indicates the update run is not started on this resource.
+	UpdateStateNotStarted UpdateState = "NotStarted"
+
+	// UpdateStateRunning indicates the update run is running on this resource.
+	UpdateStateRunning UpdateState = "Running"
+
+	// UpdateStateCompleted indicates the update run is completed on this resource.
+	UpdateStateCompleted UpdateState = "Completed"
+
+	// UpdateStateFailed indicates the update run is failed on this resource.
+	UpdateStateFailed UpdateState = "Failed"
+
+	// UpdateStateWaitingForApproval indicates the update run is waiting for approval on this resource.
+	UpdateStateWaitingForApproval UpdateState = "WaitingForApproval"
+)
+
+type AfterStageTaskStatus struct {
+	// The type of the post update task.
+	// +kubebuilder:validation:Enum=TimedWait;Approval
+	// +kubebuilder:validation:Required
+	Type AfterStageTaskType `json:"type"`
+
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	//
+	// Conditions is an array of current observed conditions for the specific type of post update task.
+	// Known conditions are "ApprovalRequestCreated", "WaitTimeElapsed", and "ApprovalRequestApproved".
+	// +kubebuilder:validation:Optional
+	Conditions []metav1.Condition `json:"conditions"`
+}
```

Comment:
Add ApprovalRequest which is the actual name of the approval request? Not sure whether you need namespace too.